src/hooks/useOfflineDetector.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useRef, useState } from 'react';
+
+export type UseOfflineDetectorParams = {
+    setStatus: (online: boolean) => void;
+    pollingDelay: number;
+    pollingUrl: string;
+};
+
+export const useOfflineDetector = ({ setStatus, pollingDelay = 5000, pollingUrl = '/' }: UseOfflineDetectorParams) => {
+    /** General status */
```

Comment:
This is not critical, but as recommendation this part of code may be changed to code with obvious names of variables without useless comments. Ex:

```ts
export const useOfflineDetector = ({ setStatus, pollingDelay = 5000, remoteServerUrl = '/' }: UseOfflineDetectorParams) => {
    const [globalOnlineStatus, setGlobalOnlineStatus] = useState(true);
    const [remoteServerStatus, setRemoteServerStatus] = useState(true);
    const remoteServerPollingInProgress = useRef(true);
    
    // ...
    
    (() => {
         if (!remoteServerPollingInProgress.current) return;
         //...
    })(); 
    
    // ...
```